### ISSUE TITLE
Allow selection on partial overlap

### ISSUE THREAD

**Issue by FriederHannenheim** [flxzt/rnote#155]
When selecting things using the freehand or square tool the stroke must be fully selected. The invisible stroke boundary is bigger than the actual stroke so you need to have some padding when selecting things. This seems unintuitive. The Apple notes app on the IPad selects things that are only partially in the selection area. This way seems better. 
**The solution**
Maybe all strokes where more than 50% of their area is in the selection zone should be selected. 
**Problems with this solution**
Big strokes could be accidentally selected if more than 50% of their area is selected. Some clever math could be done with the total are of the selection area and the stroke are to see if the selection makes sense.

**Comment by flxzt** [comment]
Can you take a recording of the behaviour in the apple notes app? I don't have an apple device.

But I think this is a good idea as an additional selector mode and should not be too hard to implement.

Also: Recently I improved the bounding /hitboxes for the strokes and shapes to make them tighter, especially for lines and ellipses and the new curve shapes. So that should improve the existing selector modes a little bit.

**Comment by FriederHannenheim** [comment]
I don't have an Apple device either I just saw one of my classmates taking notes

**Comment by PhilDevProg** [comment]
> But I think this is a good idea as an additional selector mode and should not be too hard to implement.
@flxzt I don't think that this should be a new selector, it should be an option for the rectangle and freehand selectors in my opinion that could be placed under the 4 selector modes and would be grayed out when using the 2 other selectors.

**Comment by ElbertCode** [comment]
> it should be an option for the rectangle and freehand selectors in my opinion that could be placed under the 4 selector modes and would be grayed out when using the 2 other selectors.

Maybe this could even be done with a configuration menu, where you can set the percentage overlap needed for a selection and, if implemented, a toggle whether a more complex determination algorithm should be used.

**Comment by dennisorlando** [comment]
on it
